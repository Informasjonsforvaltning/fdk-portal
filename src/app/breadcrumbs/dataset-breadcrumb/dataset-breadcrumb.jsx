import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { resolve } from 'react-resolver';
import { getDataset } from '../../../api/datasets';
import { getTranslateText } from '../../../lib/translateText';

const memoizedGetDataset = _.memoize(getDataset);

export const PureDatasetBreadcrumb = props => {
  const { datasetItem } = props;
  return <span>{getTranslateText(datasetItem && datasetItem.title)}</span>;
};

const mapProps = {
  datasetItem: props => memoizedGetDataset(props.match.params.id)
};

PureDatasetBreadcrumb.defaultProps = {
  datasetItem: null
};

PureDatasetBreadcrumb.propTypes = {
  datasetItem: PropTypes.object
};

export const DatasetBreadcrumb = resolve(mapProps)(PureDatasetBreadcrumb);
