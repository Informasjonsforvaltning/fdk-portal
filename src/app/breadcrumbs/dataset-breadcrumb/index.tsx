import React, { memo, FC } from 'react';
import { compose } from 'redux';

import { getTranslateText as translate } from '../../../lib/translateText';
import translations from '../../../lib/localization';

import withDataset, {
  Props as DatasetProps
} from '../../../components/with-dataset';

interface Props extends DatasetProps {}

const DatasetBreadcrumb: FC<Props> = ({ dataset }) => (
  <span>{translate(dataset?.title ?? translations.breadcrumb.notFound)}</span>
);

export default compose<FC>(memo, withDataset)(DatasetBreadcrumb);
