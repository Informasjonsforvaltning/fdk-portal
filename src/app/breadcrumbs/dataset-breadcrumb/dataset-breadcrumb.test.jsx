import React from 'react';
import { shallow } from 'enzyme';
import { PureDatasetBreadcrumb } from './dataset-breadcrumb';

test('should render PureDatasetBreadcrumb with "Dataset" active correctly', () => {
  const defaultProps = {
    datasetItem: {
      title: 'dataset title'
    }
  };
  const wrapper = shallow(<PureDatasetBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render PureDatasetBreadcrumb correctly with no datasetItem', () => {
  const defaultProps = {};
  const wrapper = shallow(<PureDatasetBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
