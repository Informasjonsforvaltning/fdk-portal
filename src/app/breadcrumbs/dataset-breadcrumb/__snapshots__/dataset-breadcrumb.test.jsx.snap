// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PureDatasetBreadcrumb correctly with no datasetItem 1`] = `<span />`;

exports[`should render PureDatasetBreadcrumb with "Dataset" active correctly 1`] = `
<span>
  dataset title
</span>
`;
